test/libsolidity/semanticTests/various/test_underscore_in_bin.sol

```diff
@@ -0,0 +1,10 @@
+contract test {
+    function f(bool cond) public pure returns (uint256) {
+        uint32 x = 0b0001001000110100_10101011;
```

Comment:
May I ask writing semantic tests like 
```
contract test {
    function add() public pure returns (uint c) {
        c = 0b01 + 0b110;
    }
    function sub() public pure returns (uint c) {
        c = 0b0111110 - 0b110;
    }
    function mul() public pure returns (uint c) {
        c = 0b110 * 0b01011;
    }
    function div() public pure returns (uint c) {
        c = 0b10111 / 0b11;
    }
    function mod() public pure returns (uint c) {
        c = 0b110101010 / 0b1011;
    }
}
// ----
// add() -> 0x7
// sub() -> 0x38
// mul() -> 0x42
// div() -> 0x26
```
is the appropriate way?

Reply:
Can you please point some existing tests that tests literal arithmetic, like `0x01 + 0x10` or `1 + 2`? I agree that adding tests for binary number literal is necessary because it is a new feature, but if those tests exists, referencing them would be helpful for me to add more tests.

Reply:
The above is fine; I doubt we have many such tests, since it's been a very long time since we've introduced a feature like this; you can take a look at `semanticTests/arithmetics`, `semanticTests/exponentiation`, and probably some more, but as I said, what you have here should be enough.

Reply:
I have added two tests at https://github.com/ethereum/solidity/pull/15046/commits/7cdaa37565e0d44628c3f5d3a5b120df13e2847c.